plugins/plugin-chart-echarts/src/BoxPlot/transformProps.ts

```diff
@@ -0,0 +1,178 @@
+        const groupbyLabel = extractGroupbyLabel({ datum, groupby });
+        const name = metrics.length === 1 ? groupbyLabel : `${groupbyLabel}, ${metric}`;
+        // Outlier data is a nested array of numbers (uncommon, therefore no need to add to DataRecordValue)
+        const outlierDatum = (datum[`${metric}__outliers`] || []) as number[];
+        return {
+          name: 'outlier',
+          type: 'scatter',
+          data: outlierDatum.map(val => [name, val]),
+          tooltip: {
+            formatter: (param: { data: [string, number] }) => {
+              const headline = `<p><strong>${param.data[0]}</strong></p>`;
+              const stats = `${numberFormatter(param.data[1])}`;
+              return headline + stats;
+            },
+          },
+          itemStyle: {
+            color: colorFn(groupbyLabel),
+          },
+        };
+      });
+    })
+    .flat(2);
+
+  let axisLabel;
+  if (xTicksLayout === '45°') axisLabel = { rotate: -45 };
+  else if (xTicksLayout === '90°') axisLabel = { rotate: -90 };
+  else if (xTicksLayout === 'flat') axisLabel = { rotate: 0 };
+  else if (xTicksLayout === 'staggered') axisLabel = { rotate: -45 };
+  else axisLabel = { show: true };
+
+  // @ts-ignore
+  const echartOptions: echarts.EChartOption<echarts.EChartOption.SeriesBoxplot> = {
+    grid: {
+      top: 30,
+      bottom: 30,
+      left: 20,
+      right: 20,
+      containLabel: true,
+    },
+    xAxis: {
+      type: 'category',
+      data: transformedData.map(row => row.name),
+      axisLabel,
+    },
+    yAxis: {
+      type: 'value',
+      axisLabel: { formatter: numberFormatter },
+    },
+    tooltip: {
+      trigger: 'item',
+      confine: true,
+      axisPointer: {
+        type: 'shadow',
+      },
+    },
+    series: [
+      {
+        name: 'boxplot',
+        type: 'boxplot',
+        avoidLabelOverlap: true,
+        // @ts-ignore
+        data: transformedData,
+        tooltip: {
+          formatter: param => {
+            // @ts-ignore
+            const {
+              value,
+              name,
+            }: {
+              value: [number, number, number, number, number, number, number, number, number[]];
+              name: string;
+            } = param;
+            const headline = `<p><strong>${name}</strong></p>`;
+            const stats = [
+              `upper: ${numberFormatter(value[5])}`,
+              `Q3: ${numberFormatter(value[4])}`,
+              `mean: ${numberFormatter(value[6])}`,
+              `median: ${numberFormatter(value[3])}`,
+              `Q1: ${numberFormatter(value[2])}`,
+              `lower: ${numberFormatter(value[1])}`,
+              `observations: ${numberFormatter(value[7])}`,
+            ];
+            if (value[8].length > 0) {
+              stats.push(`outliers: ${numberFormatter(value[8].length)}`);
+            }
+            return headline + stats.join('<br/>');
+          },
+        },
+      },
+      // @ts-ignore
```

Comment:
I won't be losing any sleep over these three `ts-ignore`s, but just thought I should confirm that they're not a "to do" that you wanted to pursue in this PR.

Reply:
I'm hoping to tackle these when upgrading to ECharts 5.0, as the typing system should become slightly more stable in that release.